build/accelerate_launch.py

```diff
@@ -0,0 +1,97 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Script wraps launch_training to run with accelerate for multi and single GPU cases.
+Read multiGPU configuration via environment variable `SFT_TRAINER_CONFIG_JSON_PATH`
+for the path to the JSON config file with parameters or `SFT_TRAINER_CONFIG_JSON_ENV_VAR`
+for the encoded config string to parse.
+"""
+
+# Standard
+import json
+import os
+import base64
+import pickle
+import logging
+
+# Third Party
+from accelerate.commands.launch import launch_command_parser, launch_command
+import torch
+
+
+def txt_to_obj(txt):
+    base64_bytes = txt.encode("ascii")
+    message_bytes = base64.b64decode(base64_bytes)
+    try:
+        # If the bytes represent JSON string
+        return json.loads(message_bytes)
+    except UnicodeDecodeError:
+        # Otherwise the bytes are a pickled python dictionary
+        return pickle.loads(message_bytes)
+
+
+def main():
+    LOGLEVEL = os.environ.get("LOG_LEVEL", "WARNING").upper()
+    logging.basicConfig(level=LOGLEVEL)
+
+    json_configs = {}
+    json_path = os.getenv("SFT_TRAINER_CONFIG_JSON_PATH")
+    json_env_var = os.getenv("SFT_TRAINER_CONFIG_JSON_ENV_VAR")
+
+    if json_path:
+        with open(json_path, "r", encoding="utf-8") as f:
+            json_configs = json.load(f)
+
+    elif json_env_var:
+        json_configs = txt_to_obj(json_env_var)
+
+    # parse multiGPU args
+    multi_gpu_args = []
+    if json_configs.get("multiGPU") is not None:
+        logging.info("Using multi-GPU configs: %s", json_configs.get("multiGPU"))
+        for key, val in json_configs["multiGPU"].items():
+            multi_gpu_args.append(f"--{key}")
+            multi_gpu_args.append(str(val))
+
+        # add FSDP config
+        if not json_configs.get("multiGPU").get("config_file"):
+            fsdp_filepath = os.getenv(
+                "FSDP_DEFAULTS_FILE_PATH", "/app/accelerate_fsdp_defaults.yaml"
+            )
+            if os.path.exists(fsdp_filepath):
+                logging.info("Setting accelerate config file to: %s", fsdp_filepath)
+                multi_gpu_args.append("--config_file")
+                multi_gpu_args.append(fsdp_filepath)
+
+        # add num_processes to overwrite config file set one
+        if not json_configs.get("multiGPU").get("num_processes"):
+            num_gpus = torch.cuda.device_count()
+            if num_gpus > 1:
+                logging.info("Setting accelerate num processes to %s", num_gpus)
+                multi_gpu_args.append("--num_processes")
+                multi_gpu_args.append(str(num_gpus))
+    else:
+        multi_gpu_args.append("--num_processes")
+        multi_gpu_args.append("1")
```

Comment:
Another note on usage, if multiGPU is not set as an imput param, I set `num_processes` explicitly to be 1 assuming single GPU. If this is not explicitly set, then accelerate launch will try to figure out how many GPUs are available and set that value for num_processes. So even if multiGPU is not set, it can still run multiGPU if a pod has 2 GPUs available. This requires users to be explicit when they want to run multi-GPU.

Reply:
`_validate_launch_command` is called because you already used `launch_command`, see [here](https://github.com/huggingface/accelerate/blob/4d415f2129f97da4b16f29b9bf0535fe5f2fb0e8/src/accelerate/commands/launch.py#L1053).

Reply:
@fabianlim correct me if I am wrong, are we dealing with `args.num_machines` argument at all, it should always be 1 isn't it? I thought we were not dealing with multi-node training at this point isn't it? Also, I see  @anhuong is mentioning about num_processes, isn't it?

Reply:
@kmehant my bad, I have deleted my previous comments.